### PR TITLE
DM-28589: ATDomeCsc: publish the new moveCode event

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: python/lsst/ts/ATDome/enums.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+
+__all__ = ["MoveCode"]
+
+import enum
+
+
+class MoveCode(enum.IntFlag):
```
Comment (pkubanek) [inline python/lsst/ts/ATDome/enums.py:26]: This is much better than decimals, signalling to reader this is a bitmask.

File: python/lsst/ts/ATDome/enums.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# This file is part of ts_ATDome.
```
Comment (pkubanek) [inline python/lsst/ts/ATDome/enums.py:1]: Shouldn't this be rather placed as ATDome under ts_idl subsystem (python/lsst/ts/idl/enums path)?

File: python/lsst/ts/ATDome/mock_controller.py [hunk 3]
```diff
@@ -316,14 +328,14 @@ def do_short_status(self):
             dir_code = "RL"
         if az_moving:
             if self.last_rot_right:
-                move_code += 1
+                move_code += MoveCode.AZIMUTH_POSITIVE
```
Comment (pkubanek) [inline python/lsst/ts/ATDome/mock_controller.py:331]: As move_code is a bitmask, more obvious (and also a bit quicker in CPU costs) would be to use |= operator

File: python/lsst/ts/ATDome/mock_controller.py [hunk 4]
```diff
@@ -280,22 +281,33 @@ def do_short_status(self):
...
-                else:
-                    move_code += 2 * closing_code
-            state_str = "AJAR"
-            if current_position == 0:
-                state_str = "CLOSED"
-            elif current_position == 100:
-                state_str = "OPEN"
+            current_velocity = actuator.velocity(curr_tai)
+            if current_velocity < 0:
+                move_code += closing_code
```
Comment (pkubanek) [inline python/lsst/ts/ATDome/mock_controller.py:303]: |= is more appropiate (and a bit quicker) for bitmasks.

**Review by pkubanek — APPROVED** [review]
Please see suggestions for changes and improvements. Moving enums to ts_idl would make those available to other, non ts_ATDome subsystems.

**Comment by r-owen** [comment]
Thank you for the helpful suggestions. I'll change to |=.

I am very reluctant to add the the move code enumerations to ts_xml because the move code is an internal detail, and one that may change. The the only reason I am (reluctantly) publishing it is for debugging some problems we are having. I do not want any clients to rely on it (and say so in the description in ts_xml).

**Pull request merged**